src/iuse_actor.cpp

```diff
@@ -1349,48 +1349,74 @@ void reveal_map_actor::load( const JsonObject &obj )
 void reveal_map_actor::reveal_targets( const tripoint_abs_omt &map ) const
 {
     omt_find_params params{};
-    params.search_range = radius;
+    params.search_range = { 0, radius };
+    params.search_layers = omt_find_all_layers;
     params.types = omt_types;
-    params.must_see = false;
     params.existing_only = false;
 
-    for( int z = -OVERMAP_DEPTH; z <= OVERMAP_HEIGHT; z++ ) {
-        const auto places = overmap_buffer.find_all( tripoint_abs_omt( map.xy(), z ), params );
-        for( auto &place : places ) {
-            overmap_buffer.reveal( place, 0 );
+    point_abs_om origin_om_pos;
+    point_om_omt origin_local;
+    std::tie( origin_om_pos, origin_local ) = project_remain<coords::om>( map.xy() );
```

Comment:
```suggestion
    const auto [ origin_om_pos, origin_local ] = project_remain<coords::om>( map.xy() );
```

let's use https://en.cppreference.com/w/cpp/language/structured_binding